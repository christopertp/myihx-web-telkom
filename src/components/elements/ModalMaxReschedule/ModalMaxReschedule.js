import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
import Modal from '../Modal';
import styles from './styles.scoped.css';

export default function ModalMaxReschedule(props) {
  const { onClose, open } = props;

  return (
    <Modal className={styles.root} onClose={onClose} open={open}>
      <img alt="Maximum attempt" src="/assets/grfx_status_failed.svg" />
      <h2>Sorry, you canâ€™t reschedule anymore</h2>
      <p>
        You have exceeded reschedule installation limit.
        Please contact our customer service for further information.
      </p>
      <Button onClick={onClose}>Close</Button>
      <a href="tel:147">Contact Us</a>
    </Modal>
  );
}

ModalMaxReschedule.defaultProps = {
  onClose: () => { },
  open: false,
};

ModalMaxReschedule.propTypes = {
  onClose: PropTypes.func,
  open: PropTypes.bool,
};
