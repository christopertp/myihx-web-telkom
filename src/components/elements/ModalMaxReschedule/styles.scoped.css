.root {
  text-align: center;
}

.root > img {
  height: 12.5rem;
}

.root > h2 {
  font-size: 1.25rem;
  line-height: 1.3;
}

.root > p {
  font-size: 1rem;
  line-height: 1.5;
  margin: 0.5rem 0 1rem;
}

.root > *:nth-last-child(2), .root > a:last-child {
  width: 100%;
}

.root > a:last-child {
  color: #191A1A;
  display: block;
  font-size: 0.9rem;
  font-weight: 800;
  margin-top: 0.25rem;
  padding: 0.75rem 1.5rem;
  text-align: center;
  text-transform: uppercase;
}

@media only screen and (max-width: 48em) {
  .root > img {
    height: 6.25rem;
    margin-bottom: 1rem;
  }
  .root > p {
    font-size: 0.8rem;
    margin: 0.625rem 0 1.875rem;
  }
}
