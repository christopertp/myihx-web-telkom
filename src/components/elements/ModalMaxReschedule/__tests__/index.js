import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ModalMaxReschedule from '../ModalMaxReschedule';

describe('src/components/elements/ModalMaxReschedule', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ModalMaxReschedule />);
    expect(tree).toMatchSnapshot();
    ModalMaxReschedule.defaultProps.onClose();
  });
});
