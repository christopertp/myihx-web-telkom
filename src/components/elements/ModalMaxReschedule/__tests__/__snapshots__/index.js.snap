// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/ModalMaxReschedule render 1`] = `
<Modal
  className=""
  onClose={[Function]}
  open={false}
>
  <img
    alt="Maximum attempt"
    src="/assets/grfx_status_failed.svg"
  />
  <h2>
    Sorry, you canâ€™t reschedule anymore
  </h2>
  <p>
    You have exceeded reschedule installation limit. Please contact our customer service for further information.
  </p>
  <Button
    className=""
    disabled={false}
    onClick={[Function]}
    to=""
    type="button"
    variant="red"
  >
    Close
  </Button>
  <a
    href="tel:147"
  >
    Contact Us
  </a>
</Modal>
`;
